contracts/integrations/custom/CustomIntegrationLiquity.sol

```diff
@@ -0,0 +1,232 @@
+import {LowGasSafeMath} from '../../lib/LowGasSafeMath.sol';
+import {BytesLib} from '../../lib/BytesLib.sol';
+import {ControllerLib} from '../../lib/ControllerLib.sol';
+import {IBorrowerOperations} from '../../interfaces/external/liquity/IBorrowerOperations.sol';
+import {ITroveManager} from '../../interfaces/external/liquity/ITroveManager.sol';
+
+/**
+ * @title CustomIntegrationLiquity
+ * @author adamb
+ *
+ * Custom integration for Liquity Lend and Borrow LUSD
+ */
+contract CustomIntegrationLiquity is CustomIntegration {
+    uint constant LIQUITY_MAX_FEE_PERCENTAGE = 0.01 * 1e18; // 1%
+
+    using LowGasSafeMath for uint256;
+    using PreciseUnitMath for uint256;
+    using BytesLib for uint256;
+    using ControllerLib for IBabController;
+
+    /* ============ State Variables ============ */
+
+    /* Add State variables here if any. Pass to the constructor */
+    IBorrowerOperations private immutable liquityBorrowerOperations;
+
+    /* ============ Constructor ============ */
+
+    /**
+     * Creates the integration
+     *
+     * @param _controller                   Address of the controller
+     */
+    constructor(IBabController _controller, IBorrowerOperations _liquityBorrowerOperations) CustomIntegration('custom_liquity', _controller) {
+        require(address(_controller) != address(0), 'invalid address');
+        liquityBorrowerOperations = _liquityBorrowerOperations;
+    }
+
+    /* =============== Internal Functions ============== */
+
+    /**
+     * Whether or not the data provided is valid
+     *
+     * hparam  _data                     Data provided
+     * @return bool                      True if the data is correct
+     */
+    function _isValid(bytes memory _data) internal view override returns (bool) {
+        (, uint amountLUSDToBorrow) = abi.decode(_data, (address, uint));
+        return amountLUSDToBorrow >= liquityBorrowerOperations.MIN_NET_DEBT();
+    }
+
+    /**
+     * Which address needs to be approved (IERC-20) for the input tokens.
+     *
+     * hparam  _data                     Data provided
+     * hparam  _opType                   O for enter, 1 for exit
+     * @return address                   Address to approve the tokens to
+     */
+    function _getSpender(
+        bytes calldata, /* _data */
+        uint8 /* _opType */
+    ) internal view override returns (address) {
+        // No approval needed, as we are lending ETH
+        // Can't use address(0), so just use the borrower operations
+        return address(liquityBorrowerOperations);
+    }
+
+    /**
+     * The address of the IERC-20 token obtained after entering this operation
+     *
+     * hparam  _token                     Address provided as param
+     * @return address                    Address of the resulting lp token
+     */
+    function _getResultToken(address) internal view override returns (address) {
+        return liquityBorrowerOperations.lusdToken();
```

Comment:
It's a little weird to have LUSD as the result token, since it doesn't reflect the amount of ETH that we deposited to our trove. Is this OK?

Reply:
Yes this is fine but when we return the price of the result token we need to account for the value of the collateral.  See below